homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -1,12 +1,294 @@
+        )
+        self._device_entries = device_entries
+
+        devices = {
+            entry.id: entry.name_by_user if entry.name_by_user else entry.name
+            for entry in device_entries
+        }
+
+        options = {
+            vol.Optional(CONF_DEBUG, default=self._config_entry.data[CONF_DEBUG]): bool,
+            vol.Optional(
+                CONF_AUTOMATIC_ADD, default=self._config_entry.data[CONF_AUTOMATIC_ADD],
+            ): bool,
+            vol.Optional(CONF_EVENT_CODE): str,
+            vol.Optional(CONF_DEVICE): vol.In(devices),
+            vol.Optional(CONF_REMOVE_DEVICE): vol.In(devices),
+        }
+
+        return self.async_show_form(
+            step_id="prompt_options", data_schema=vol.Schema(options), errors=errors
+        )
+
+    async def async_step_set_device_options(self, user_input=None):
+        """Manage device options."""
+        errors = {}
+
+        if user_input is not None:
+            device_id = get_device_id(
+                self._selected_device_object.device,
+                data_bits=user_input.get(CONF_DATA_BITS),
+            )
+            try:
+                command_on = (
+                    int(user_input.get(CONF_COMMAND_ON), 16)
+                    if user_input.get(CONF_COMMAND_ON)
+                    else None
+                )
+                command_off = (
+                    int(user_input.get(CONF_COMMAND_OFF), 16)
+                    if user_input.get(CONF_COMMAND_OFF)
+                    else None
+                )
+            except (NameError, ValueError):
+                errors = {"base": "invalid_input_2262"}
+
+            if not errors:
+                devices = {}
+                device = {
+                    CONF_DEVICE_ID: device_id,
+                    CONF_FIRE_EVENT: user_input.get(CONF_FIRE_EVENT, False),
+                    CONF_SIGNAL_REPETITIONS: user_input.get(CONF_SIGNAL_REPETITIONS, 1),
+                }
+
+                devices[self._selected_device_event_code] = device
+
+                if (
+                    user_input.get(CONF_OFF_DELAY_ENABLED)
+                    and user_input.get(CONF_OFF_DELAY) is not None
+                ):
+                    device[CONF_OFF_DELAY] = user_input[CONF_OFF_DELAY]
+                if user_input.get(CONF_DATA_BITS):
+                    device[CONF_DATA_BITS] = user_input.get(CONF_DATA_BITS)
+                if command_on:
+                    device[CONF_COMMAND_ON] = command_on
+                if command_off:
+                    device[CONF_COMMAND_OFF] = command_off
+
+                self.update_config_data(
+                    global_options=self._global_options, devices=devices
+                )
+
+                return self.async_create_entry(title="", data={})
+
+        device_data = self._selected_device
+
+        data_scheme = {
+            vol.Optional(
+                CONF_FIRE_EVENT, default=device_data.get(CONF_FIRE_EVENT, False)
```

Comment:
Why do we need an option for fire event? We could always fire an event.

Reply:
@elupus What do you think? Does it hurt to fire an event for all devices that are in the device list?

Reply:
From my perspective no, but balloob didn't want it. I think we should, it's not many events and it's much easier for user.

Reply:
@MartinHjelmare See #37565

I think in time, this event can go, as binary sensor and sensor have force_update now so they always fire state events when receiving data. We have however seen that some commands do not end up in entities, like Group On/Off. Before we remove this, we need to be sure that events are always mapped to an entity.

Perhaps not touch it now and see if we can improve event to entity mapping such that it can be removed altogether at a later point in time?

Reply:
Then we can question the need for binary sensor all together. But it is starting to get tricky for user to configure this. They need to know quite intricate details of rfxtrx.